packages/web-pkg/src/components/Avatars/UserAvatar.vue

```diff
@@ -0,0 +1,47 @@
+<template>
+  <oc-avatar :user-name="userName" :src="avatarSrc" :width="36" />
+</template>
+
+<script setup lang="ts">
+import { computed, onBeforeUnmount, onMounted, unref } from 'vue'
+import { useAvatarsStore, useClientService } from '../../composables'
+import { storeToRefs } from 'pinia'
+
+const { userId } = defineProps<{
+  userId: string
+  userName: string
+}>()
+
+const avatarsStore = useAvatarsStore()
+const { avatarMap } = storeToRefs(avatarsStore)
+const clientService = useClientService()
+
+const controller = new AbortController()
+
+const avatarSrc = computed(() => {
+  return unref(avatarMap)[userId]
+})
+
+const loadAvatar = async () => {
```

Comment:
When you extract this to a composable in the future as you mentioned, could you please make this a task? Then you can just cancel it on unmount, no need for the `AbortController`.

Reply:
👍 ofc